api/tests/Feature/CandidateAssessmentStatusTest.php

```diff
@@ -504,4 +505,124 @@ public function testOutOfOrderAssessmentsDoNotIncrementStep()
                 ],
             ]);
     }
+
+    public function testMultipleEssentialMustPassToIncrementStep()
+    {
+
+        $pool = Pool::factory()
+            ->published()
+            ->create([
+                'team_id' => $this->team->id,
+            ]);
+
+        $technicalSkills = Skill::where('category', SkillCategory::TECHNICAL->name)->limit(2)->get();
+        $poolSkillOne = PoolSkill::create([
+            'pool_id' => $pool->id,
+            'skill_id' => $technicalSkills[0]->id,
+            'type' => PoolSkillType::ESSENTIAL->name,
+        ]);
+
+        // Non-essential skills must also be assessed if a user claims them
+        $poolSkillTwo = PoolSkill::create([
+            'pool_id' => $pool->id,
+            'skill_id' => $technicalSkills[1]->id,
+            'type' => PoolSkillType::NONESSENTIAL->name,
+        ]);
+
+        $user = User::factory()->create();
+        UserSkill::factory()->create([
+            'user_id' => $user->id,
+            'skill_id' => $poolSkillTwo->skill_id,
+        ]);
+
+        $candidate = PoolCandidate::factory()->withSnapshot()->create([
+            'user_id' => $user->id,
+            'pool_id' => $pool->id,
+            'submitted_at' => config('constants.past_date'),
+            'expiry_date' => config('constants.far_future_date'),
+        ]);
+
+        $stepOne = $pool->assessmentSteps->first();
+
+        $stepTwo = AssessmentStep::factory()
```

Comment:
I don't know if it matters, but I feel like you should create the assessment plan before creating the application with snapshot.

Reply:
Yeah, not necessary but makes the test easier to follow so I updated the order in [adjust test names, order of creation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10925/commits/db93cca5019117064fe22794f3e8f8da2267b2a3)